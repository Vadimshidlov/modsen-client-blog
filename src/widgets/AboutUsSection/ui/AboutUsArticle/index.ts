export { AboutUsArticle } from '@/widgets/AboutUsSection/ui/AboutUsArticle/AboutUsArticle';
