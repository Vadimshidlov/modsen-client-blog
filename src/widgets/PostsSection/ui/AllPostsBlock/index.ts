export { AllPostsBlock } from '@/widgets/PostsSection/ui/AllPostsBlock/AllPostsBlock';
