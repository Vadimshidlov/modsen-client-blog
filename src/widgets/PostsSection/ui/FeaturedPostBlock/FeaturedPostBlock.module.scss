.section__container {
  max-width: 733px;

  @media (max-width: 1380px) {
    max-width: 100%;
    margin-bottom: 20px;
  }

  .post__container {
    display: flex;
    flex-direction: column;
    padding: 32px;
    border: 1px solid $grey;

    .post__image {
      width: 100%;
      object-fit: cover;
      margin-bottom: 32px;
    }

    .post__authorInfo {
      font-size: $fz16;
      font-weight: $regular;
      margin-bottom: 16px;
    }

    .post__title {
      font-size: $fz28;
      font-weight: $bold;
      line-height: 40px;
    }

    .post__text {
      font-size: $fz16;
      font-weight: $regular;
      line-height: 28px;
      margin-bottom: 32px;
    }
  }
}


.section__title {
  font-size: $fz36;
  font-weight: $bold;
  line-height: 48px;
  margin-bottom: 32px;

  @media (max-width: 768px) {
    font-size: $fz28;
    margin-bottom: 16px;
  }

  @media (max-width: 450px) {
    font-size: $fz22;
    margin-bottom: 10px;
  }
}
