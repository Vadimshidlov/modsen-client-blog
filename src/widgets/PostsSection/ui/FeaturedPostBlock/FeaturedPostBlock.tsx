import React from 'react';
import styles from '@/widgets/PostsSection/ui/FeaturedPostBlock/FeaturedPostBlock.module.scss';
import Image from 'next/image';
import { Button } from '@/shared/ui/Button/Button';
import { useTranslations } from 'next-intl';
import Link from 'next/link';
import { POSTS } from '@/shared/constants/posts';

export function FeaturedPostBlock() {
  const t = useTranslations('FeaturedPostBlock');

  const featuredPostData = POSTS[POSTS.length - 1];
  const { id, author, postDate, title, text, img } = featuredPostData;

  return (
    <section className={styles.section__container}>
      <h3 className={styles.section__title}>{t('title')}</h3>
      <div className={styles.post__container}>
        <Image className={styles.post__image} alt="Mountains" src={img} placeholder="blur" />
        <span className={styles.post__authorInfo}>
          {t('authorTag')}
          &nbsp;{author} {postDate}
        </span>
        <h4 className={styles.post__title}>{title}</h4>
        <span className={styles.post__text}>{text}</span>
        <Link href={`/posts/${id}`}>
          <Button>{t('buttonText')} &gt;</Button>
        </Link>
      </div>
    </section>
  );
}
