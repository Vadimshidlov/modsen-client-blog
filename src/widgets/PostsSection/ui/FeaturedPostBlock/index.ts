export { FeaturedPostBlock } from '@/widgets/PostsSection/ui/FeaturedPostBlock/FeaturedPostBlock';
