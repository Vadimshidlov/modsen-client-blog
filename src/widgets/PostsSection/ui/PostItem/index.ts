export { PostItem } from '@/widgets/PostsSection/ui/PostItem/PostItem';
