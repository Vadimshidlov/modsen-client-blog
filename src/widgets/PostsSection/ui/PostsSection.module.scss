.posts {
  padding-top: 128px;
  margin-bottom: 128px;
  display: flex;
  justify-content: space-between;
  column-gap: 28px;

  @media (max-width: 1380px) {
    flex-direction: column;
    align-items: center;
  }

  @media (max-width: 768px) {
    padding-top: 64px;
    margin-bottom: 64px;
  }

  @media (max-width: 450px) {
    padding-top: 32px;
    margin-bottom: 32px;
  }
}
