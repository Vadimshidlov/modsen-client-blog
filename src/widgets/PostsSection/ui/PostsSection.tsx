import React from 'react';
import { FeaturedPostBlock } from '@/widgets/PostsSection/ui/FeaturedPostBlock';
import { AllPostsBlock } from '@/widgets/PostsSection/ui/AllPostsBlock';
import styles from '@/widgets/PostsSection/ui/PostsSection.module.scss';
import container from '@/shared/styles/container.module.scss';

export function PostsSection() {
  return (
    <section className={`${styles.posts} ${container.container}`}>
      <FeaturedPostBlock />
      <AllPostsBlock />
    </section>
  );
}
