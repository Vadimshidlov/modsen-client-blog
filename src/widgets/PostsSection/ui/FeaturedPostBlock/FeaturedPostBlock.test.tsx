import React from 'react';
import { render } from '@testing-library/react';
import { FeaturedPostBlock } from '@/widgets/PostsSection/ui/FeaturedPostBlock';

jest.mock('next-intl', () => ({
  useTranslations: jest.fn().mockReturnValue((key: string) => `Mocked ${key}`),
}));

describe('FeaturedPostBlock tests', () => {
  test('should matches snapshot', () => {
    const { container } = render(<FeaturedPostBlock />);
    expect(container).toMatchSnapshot();
  });
});
