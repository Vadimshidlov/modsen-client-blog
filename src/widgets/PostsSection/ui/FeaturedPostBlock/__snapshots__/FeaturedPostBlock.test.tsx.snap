// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturedPostBlock tests should matches snapshot 1`] = `
<div>
  <section
    class="section__container"
  >
    <h3
      class="section__title"
    >
      Mocked title
    </h3>
    <div
      class="post__container"
    >
      <img
        alt="Mountains"
        class="post__image"
        data-nimg="1"
        decoding="async"
        height="40"
        loading="lazy"
        src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
        style="color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
        width="40"
      />
      <span
        class="post__authorInfo"
      >
        Mocked authorTag
         
        Robert Fox
         
        July 20, 2024
      </span>
      <h4
        class="post__title"
      >
        Rabbit’s R1 is a little AI gadget that grows on you: 
      </h4>
      <span
        class="post__text"
      >
        Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident.
      </span>
      <a
        href="/posts/9"
      >
        <button
          class="button"
          type="button"
        >
          Mocked buttonText
           &gt;
        </button>
      </a>
    </div>
  </section>
</div>
`;
